src/Sylius/Bundle/CoreBundle/Migrations/Utility/TopologyBasedVersionComparator.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Migrations\Utility;
```

Comment:
```suggestion
namespace Sylius\Bundle\CoreBundle\Migrations\Comparator;
```
or
```suggestion
namespace Sylius\Bundle\CoreBundle\Doctrine\Migrations\Comparator;
```

The `utility` namespace can be a pandora box 📦

Reply:
The idea is to either extract it to a standalone package or push those changes upstream.